### PR TITLE
Shinyngs/validator: Make feature file optional

### PR DESCRIPTION
**Pull request by WackerO**
This PR makes the feature file an optional input to validator and splits the output into separate channels

## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Review by pinin4fjords — CHANGES_REQUESTED** [review]
In addition to the other changes, could we also have a new test where the features are not supplied?

File: modules/nf-core/shinyngs/validatefomcomponents/main.nf [hunk 1]
```diff
@@ -24,15 +24,19 @@ process SHINYNGS_VALIDATEFOMCOMPONENTS {
...
     """
     validate_fom_components.R \\
         --sample_metadata $sample \\
-        --feature_metadata $feature_meta \\
+        $feature \\
         --assay_files ${assay_files.join(',')} \\
         --contrasts_file $contrasts \\
         --output_directory $prefix \\
         $args
+    if [ -z $feature ]
```
[the remark this answers is not in the document]
Comment (pinin4fjords) [reply modules/nf-core/shinyngs/validatefomcomponents/main.nf:36]: @WackerO actually, there's a previously undocumented (but [recently documented](https://github.com/nf-core/nf-co.re/pull/1461) after I asked in response to this) convention that each output file should be in a separate channel. Would you be able to to do that as part of this PR? Then the features output can simply be marked as optional.

File: modules/nf-core/shinyngs/validatefomcomponents/main.nf [hunk 2]
```diff
@@ -8,12 +8,16 @@ process SHINYNGS_VALIDATEFOMCOMPONENTS {
         'quay.io/biocontainers/r-shinyngs:1.5.1--r42hdfd78af_0' }"
 
     input:
-    tuple val(meta), path(sample), path(feature_meta), path(assay_files)
-    tuple val(meta2), path(contrasts)
+    tuple val(meta),  path(sample)
```
Comment (pinin4fjords) [inline modules/nf-core/shinyngs/validatefomcomponents/main.nf:11]: Can we leave the samples and assays in the same input channel? That makes sense I think

Comment (WackerO) [reply modules/nf-core/shinyngs/validatefomcomponents/main.nf]: done!

Comment (WackerO) [reply modules/nf-core/shinyngs/validatefomcomponents/main.nf]: actually, I am just now wondering if there is a reason why you added the meta2 to contrasts? The metamap is always the same, isn't it? If so I would just have it input into the module in the first input channel, and the other input is just provided without the `tuple val(meta)` thing?

Comment (pinin4fjords) [reply modules/nf-core/shinyngs/validatefomcomponents/main.nf]: No, these are representing different things, the contrast meta has different content to the samples/assays meta.

Comment (pinin4fjords) [reply modules/nf-core/shinyngs/validatefomcomponents/main.nf]: In the specific test examples the meta is the same, but I wouldn't want to impose that assumption.

File: modules/nf-core/shinyngs/validatefomcomponents/meta.yml [hunk 3]
```diff
@@ -23,6 +23,16 @@ input:
       description: |
         Groovy Map containing information on experiment, at a minimum an id. To match meta.
```
Comment (pinin4fjords) [inline modules/nf-core/shinyngs/validatefomcomponents/meta.yml:24]:
```suggestion
        Groovy Map containing information on features.
```

File: modules/nf-core/shinyngs/validatefomcomponents/meta.yml [hunk 4]
```diff
@@ -23,6 +23,16 @@ input:
...
         e.g. [ id:'test' ]
+  - meta3:
+      type: map
+      description: |
+        Groovy Map containing information on experiment, at a minimum an id. To match meta.
```
Comment (pinin4fjords) [inline modules/nf-core/shinyngs/validatefomcomponents/meta.yml:29]:
```suggestion
        Groovy Map containing information on contrasts.
```

File: modules/nf-core/shinyngs/validatefomcomponents/meta.yml [hunk 5]
```diff
@@ -23,6 +23,16 @@ input:
...
+        e.g. [ id:'test' ]
+  - meta4:
+      type: map
+      description: |
+        Groovy Map containing information on experiment, at a minimum an id. To match meta.
+        e.g. [ id:'test' ]
```
Comment (pinin4fjords) [inline modules/nf-core/shinyngs/validatefomcomponents/meta.yml:35]:
```suggestion
```

File: modules/nf-core/shinyngs/validatefomcomponents/meta.yml [hunk 6]
```diff
@@ -50,15 +60,11 @@ output:
       type: file
       description: File containing validated sample metadata
       pattern: "/*.sample_metadata.tsv"
-  - sample_meta:
-      type: file
-      description: File containing validated feature metadata
-      pattern: "/*.feature_metadata.tsv"
-  - asays:
+  - assays:
```
Comment (pinin4fjords) [inline modules/nf-core/shinyngs/validatefomcomponents/meta.yml:63]:
```suggestion
  - feature_meta:
      type: file
      description: File containing validated feature metadata
      pattern: "/*.feature_metadata.tsv"
      optional: true
  - assays:
```

Sorry I obviously did a poor job of this originally!

**Review by pinin4fjords — APPROVED** [review]
LGTM!

**Pull request merged**